fuzzbench/run_experiment.py

```diff
@@ -17,37 +17,72 @@
 
 import redis
 import rq
+from rq.job import Job
 
+from common import config_utils, environment, yaml_utils
+from experiment.build import docker_images
 from fuzzbench import jobs
 
+redis_connection = redis.Redis(host="queue-server") # pylint: disable=invalid-name
 
-def run_experiment():
+
+def run_experiment(config):
     """Main experiment logic."""
     print('Initializing the job queue.')
     queue = rq.Queue()
+
+    images_to_build = docker_images.get_images_to_build(config['fuzzers'],
+                                                        config['benchmarks'])
     jobs_list = []
-    jobs_list.append(
-        queue.enqueue(jobs.build_image,
-                      'base-image',
-                      job_timeout=600,
-                      job_id='base-image'))
-    jobs_list.append(
-        queue.enqueue(jobs.build_image,
-                      'base-builder',
-                      job_timeout=600,
-                      job_id='base-builder',
-                      depends_on='base-image'))
-    jobs_list.append(
-        queue.enqueue(jobs.build_image,
-                      'base-runner',
-                      job_timeout=600,
-                      job_id='base-runner',
-                      depends_on='base-image'))
+    unqueued_build_images = []
+    for name, obj in images_to_build.items():
+        if name in ['base-image']:
```

Comment:
why does base image need a separate branch?

Reply:
>why does base image need a separate branch?

This image is the only one not depending on other images. We can also write as
```
if obj.get('depends_on', None) is None:
```
Wdyt? Now it seems we change to this way.

Reply:
Does it actually matter what order do we enqueue jobs? E.g., would that be an issue if we happen to first
- enqueue('base-builder', depends_on='base_image), and then
- enqueue('base_image)

?

Reply:
Good question! Let's me have a try for this.